webpack/scenes/RedHatRepositories/components/RepositorySetRepositories.js

```diff
@@ -28,7 +28,36 @@ class RepositorySetRepositories extends Component {
     }
 
     const availableRepos = [...data.repositories.filter(({ enabled }) => !enabled)]
```

Comment:
@johnpmitsch any advice on how to test `availableRepos`? Since it's effectively private my first thought would be to test the output of `render` itself - but want to make sure that is correct since this test relies on external data that we would mock out.

Ultimately we are wanting to test that the RepositorySetRepository components are in the correct order once rendered .. something like:

```
<RepositorySetRepositories>
  <RepositorySetRepository ...>
  <RepositorySetRepository ...>
  <RepositorySetRepository ...>
</RepositorySetRepository>
```

Does `shallow` make this an impossiblity?

Reply:
A few thoughts:

This feels like something that should be outside of render.

I'm thinking something like this method declared in the class (all untested):
```javascript
sortedRepos = () => {
  const { data } = this.props;
  return [...data.repositories.filter(({ enabled }) => !enabled)]
    .sort((repo1, repo2) => {
      //sort logic
    }
}
```

Then in render() do this:
```javascript
      const availableRepos = sortedRepos().map(repo => (
        <RepositorySetRepository key={repo.arch + repo.releasever} type={type} {...repo} />
      ));

    const repoMessage = (data.repositories.length > 0 && availableRepos.length === 0 ?
      __('All available architectures for this repo are enabled.') : __('No repositories available.'));

    return (
      <Spinner loading={data.loading}>
        {availableRepos.length ? availableRepos : <div>{repoMessage}</div>}
      </Spinner>
    );
```

Then you can test the sortedRepos() function by using the snapshot instance. You may have to modify it to take arguments rather than read from props. There is an example [here](https://github.com/Katello/katello/blob/master/webpack/scenes/Subscriptions/UpstreamSubscriptions/__tests__/UpstreamSubscriptionsPage.test.js#L38) of testing [this method](https://github.com/Katello/katello/blob/master/webpack/scenes/Subscriptions/UpstreamSubscriptions/UpstreamSubscriptionsPage.js#L54)